src/subtitle/cue.ts

```diff
@@ -0,0 +1,164 @@
+}
+
+/**
+ * Represents the type value of value that should be considered for a line or position value.
+ */
+export enum CueValueType {
+  /**
+   * Value is automatically set so the number associated with line/position can be ignored.
+   */
+  AUTO = 'auto',
+  /**
+   * Value isn't automatically set so the number associated with line/position should be considered.
+   */
+  NUMERIC = 'numeric',
+}
+
+/**
+ * Represents the vertical position of the cue box.
+ */
+export interface CueLine {
+  /**
+   * Alignment for the line property.
+   */
+  align: CueAlignment;
+  /**
+   * The line value type. Can be either `AUTO` or `NUMERIC`, for an actual numeric value.
+   */
+  type: CueValueType;
+  /**
+   * The actual value for the line property.
+   * If type is set to `AUTO`, then this value should be ignored.
+   */
+  value: number;
+}
+
+/**
+ * Represents the horizontal position of the cue box.
+ */
+export interface CueVttPosition {
+  /**
+   * Alignment for the position property.
+   */
+  align: CueAlignment;
+  /**
+   * The position value type. Can be either `AUTO` or `NUMERIC`, for an actual numberic value.
+   */
+  type: CueValueType;
+  /**
+   * The actual value for the position property.
+   * If type is set to `AUTO`, then this value should be ignored.
+   */
+  value: number;
+}
+
+/**
+ * Represents the positioning information for CEA captions.
+ */
+export interface CueCeaPosition {
+  /**
+   * Index of the row ranging from 0 to 14.
+   */
+  row: number;
+  /**
+   * Index of the column ranging from 0 to 31.
+   */
+  column: number;
+}
+
+/**
+ * Represents a cue of a subtitle track.
+ */
+export interface Cue {
+  /**
+   * Direction in which the cue's text should be rendered.
+   */
+  direction?: CueDirection;
+  /**
+   * The end time of the cue in seconds.
+   */
+  end: number;
+  /**
+   * The cue text as HTML.
+   */
+  html?: string;
+  /**
+   * The vertical position of the cue box.
+   */
+  line?: CueLine;
+  /**
+   * The position of the cue for CEA-captions only.
+   *
+   * Provides information about where the cue should be positioned on a grid of 15 character rows
+   * times 32 columns.
+   *
+   * Only available for iOS.
+   */
+  ceaPosition?: CueCeaPosition;
+  /**
+   * The VTT positioning properties for this cue.
+   */
+  vttPosition?: CueVttPosition;
+  /**
+   * The size of the cue box relative to the video dimensions.
+   *
+   * @remarks
+   * If the writing direction is horizontal, then the size percentages are relative to the width of
+   * the video, otherwise to the height of the video.
+   */
+  size?: number;
+  /**
+   * The start time of the cue in seconds.
+   */
+  start: number;
+  /**
+   * The cue text.
+   */
+  text?: string;
+  /**
+   * The anchor point in which text should be rendered.
+   */
+  textAlignment?: CueAlignment;
+  /**
+   * The cue image data as a base64 png source.
+   * Can be passed to the `<Image />` component as the `src` prop.
+   */
+  image?: string;
+}
```

Comment:
Yes, initially I was thinking about the possibility of the end-user rendering their own cues from RN, since this will be a common use-case considering that maybe custom UI will be supported [soon](https://github.com/bitmovin/bitmovin-player-react-native/pull/13).

Also, I know this will be needed for sure if we intend to support tvOS someday, since `WebView` based UI is not supported.

What exactly do you think is not relevant here? I could shrink this API to provide only the minimal properties for cue rendering.

Reply:
https://github.com/bitmovin/bitmovin-player-react-native/pull/13
If I undestand the PR concern correctly, this is about the configuration the mobile SDKs provide to swap out the WebUI they are using. So no `Cue` is needed on the react side as still everything will be handled by the underlying SDKs.

> Also, I know this will be needed for sure if we intend to support tvOS someday, since WebView based UI is not supported.

We can also think of a good solution once we are actively work on the tvOS support 🙂 

> What exactly do you think is not relevant here? I could shrink this API to provide only the minimal properties for cue rendering.

Basically the same discussion we have in the other thread with @rolandkakonyi. This is a mix of android and iOS API, but we actually don't need it and while strongly related to the subtitles API is actually a different concern.

Reply:
Alright, got it. No problems then. I'll drop cue events for now, leaving just side loading subtitles and their events. Then we can get back to cues rendering later on. Thanks for your help.

Reply:
As discussed earlier, cues support has been dropped for now: 1599353362e30aa2471b60d7ddcff02d88bf8b2c